src/consensus/cfund.cpp

```diff
@@ -767,94 +887,97 @@ void CFund::CFundStep(const CValidationState& state, CBlockIndex *pindexNew, con
 
             CBlockIndex* pblockindex = mapBlockIndex[prequest->txblockhash];
 
-            CProposal proposal;
-
-            if (!view.GetProposal(prequest->proposalhash, proposal))
-                continue;
+            bool fUpdate = false;
 
             int nCreatedOnCycle = (pblockindex->nHeight / Params().GetConsensus().nBlocksPerVotingCycle);
             int nCurrentCycle = (pindexNew->nHeight / Params().GetConsensus().nBlocksPerVotingCycle);
             int nElapsedCycles = std::max(nCurrentCycle - nCreatedOnCycle, 0);
             int nVotingCycles = std::min(nElapsedCycles, (int)Params().GetConsensus().nCyclesPaymentRequestVoting + 1);
 
-            auto oldState = prequest->fState;
             auto oldCycle = prequest->nVotingCycle;
 
-            if((prequest->fState == CFund::NIL || fUndo) && nVotingCycles != prequest->nVotingCycle)
+            CPaymentRequest tmp;
+            CPaymentRequest oldprequest = CPaymentRequest();
+
+            if (fLog)
             {
-                prequest->nVotingCycle = nVotingCycles;
-                fUpdate = true;
+                view.GetPaymentRequest(it->first, tmp);
+                tmp.swap(oldprequest);
             }
 
-            if((pindexNew->nHeight + 1) % Params().GetConsensus().nBlocksPerVotingCycle == 0)
+            CProposal proposal;
+
+            if (!view.GetProposal(prequest->proposalhash, proposal))
```

Comment:
i can see it's moved here. what was the motive of moving it here? so we get the logging?

Reply:
CFundStep has been rewritten a bit, i'm not sure if this check would make more sense earlier. it's less likely a payment request has no parent proposal than a payment request being out of chain (previous checks)